pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -81,6 +88,8 @@
          * @throws Exception if building the filter failed
          */
         public Object build(Operation operation, Object left, Object right) throws Exception;
+        public Object build(LogicalOperation operation, Object left, Object right) throws Exception;
+        public Object build(LogicalOperation operation, Object filter) throws Exception;
```

Comment:
This is confusing -- a LogicalOperation can be assumed a type of Operation, but it is not -- a different Enum and no inheritance. Why do we need such a distinction ? 

If we really need to keep track of whether an operation is logical or not, have a boolean flag on the enum value.

If we consolidate, we can have just one Enum, with numbers for mapping and a flag for logical (if needed). This will reduce number of functions / enums and code overall.